<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<title>The canonical format for Linux kernel patches</title>
<link rel="stylesheet" type="text/css" href="file:///home/rene/github/github/notes/out.html//notes.css">
<script src='file:///home/rene/github/github/notes/out.html//q.js'></script>
</head>
<body>
Search notes: <input size='50' id='q' onchange='q();'>
<h1>The canonical format for Linux kernel patches</h1>
<div class='t'>
This format is defined in <a href='file:///home/rene/github/github/notes/out.html//Linux/kernel/source/Documentation/SubmittingPatches.html'>Documentation/SubmittingPatches</a> </div><div class='g'></div><div class='t'>
The patch is sent to the corresponding <a href='file:///home/rene/github/github/notes/out.html//Linux/kernel/source/MAINTAINERS.html'>maintainers</a> with such a <i>Subject</i> format:  <code>[PATCH] subsystem: summarizing phrase</code>. </div><div class='g'></div><div class='t'>
The message body looks like </div><pre class='code'>from: John Doe &lt;jd@somewhere.xyz&gt;

A commit message. This message will be added to the permanent changelog
and describes the patch. It is seperated by an empty line from the
preceeding from: line.

The from line is only needed if the person sending the patch is different
from the sender's email address. This person will be credited as the
author of the patch.

The message is followed by one or more "signed off" lines which then
is followed by a line containing three hyphens.

Signed-off-by: Donald Duck &lt;dd@disneyland.com&gt;
---
The optional text after the three hyphens won't go into the changelog.
It can be used for additional comments etc.
It is followed by the actual diff output:

  arch/arm/boot/dts/keystone.dtsi | 8 ++++++++
  1 file changed, 8 insertions(+)

diff --git a/arch/arm/boot/dts/keystone.dtsi b/arch/arm/boot/dts/keystone.dtsi
index 3f272826f537..007648971744 100644
--- a/arch/arm/boot/dts/keystone.dtsi
--- b/arch/arm/boot/dts/keystone.dtsi

   etc
   etc
--
2.1.0
</pre><div class='g'></div><div class='t'>
The <i>summarizing phrase</i> from the <i>Subject</i> becomes a globally-unique identitifier for that patch. It should not be longer than 75 characters. </div>
<div class='h'><h2>See also</h2>
<div class='t'>
  <a class='ext' href='http://linux.yyz.us/patch-format.html'>http://linux.yyz.us/patch-format.html</a> </div><div class='g'></div><div class='t'>
  <a href='file:///home/rene/github/github/notes/out.html//Linux/kernel/source/Documentation/email-clients_txt.html'>Documentation/email-clients.txt</a> </div><div class='g'></div><div class='t'>
  <a class='ext' href='https://www.ozlabs.org/~akpm/stuff/tpp.txt'>https://www.ozlabs.org/~akpm/stuff/tpp.txt</a> </div><div class='g'></div><div class='t'>
  <a class='ext' href='https://lkml.org/lkml/2005/4/7/183'>https://lkml.org/lkml/2005/4/7/183</a> </div>
</div><hr><p><a href='file:///home/rene/github/github/notes/out.html//index.html'>Index</a><div class='bottom'></div></body>
</html>